main.go

```diff
@@ -42,6 +49,28 @@ func main() {
 	log.SetFlags(0)
 	log.SetPrefix("")
 
+	// Detect if the output goes to TTY or some file
+	var isTTY bool
+	_, err := unix.IoctlGetWinsize(int(os.Stdout.Fd()), unix.TIOCGWINSZ)
+	if err != nil {
```

Comment:
I believe this can be simplified to

```go
_, err := unix.IoctlGetWinsize(int(os.Stdout.Fd()), unix.TIOCGWINSZ)
isTTY := err == nil
```

You'll also probably want to report this error, if it is not nil.

Reply:
Yeah, it can be simplified like this. I don't think that I want to report the error (`err`) in this case, because it is absolutely normal, when somebody redirects `stdout` to some file. Or do I miss anything?

Reply:
What sorts of errors do `IoctlGetWinsize` return? Are they fatal to the operation of the program or useful to report to the user? We do appear to be using it exclusively to check whether we're running in a TTY or not, so I can see the reason for not reporting it.

Reply:
I was thinking about this more, and maybe we should use something more reliable... What about using this?

https://pkg.go.dev/github.com/mattn/go-isatty#section-readme
GitHub repository of this project https://github.com/mattn/go-isatty

Reply:
Maybe using `go-isatty` is overkill in our case, because we need support only for Linux (not for Plan9, Solaris, etc.) We can grab 3 lines of code from there and just use it.

```go
func IsTerminal(fd uintptr) bool {
	_, err := unix.IoctlGetTermios(int(fd), unix.TCGETS)
	return err == nil
}
```

Reply:
Yea. I don't mean to diminish the value that package; under certain conditions, I can agree with abstracting that kind of functionality into a package. But for our use here, calling `unix.IoctlGetTermios` directly seems sufficient.

Reply:
Funny, I should reload page before replying to any comment and don't trust to automatic updating of GitHub webpage ... It seems we had same idea.